### PR TITLE
(MAINT) pl:fetch is ☠️

### PR DESCRIPTION
**Pull request by mckern**
The way we use a Git repo as a data service is a **bad time**, but we don't need to make it worse by only being able to access it from a Rake task. The `BuildParams` class encapsulates all of the logic that was previously wrapped up in a Rake task, and is now required alongside the `Pkg::Config` class whenever `packaging` is required.

As a side effect, `pl:load_extras` no longer exists and is no longer required to get the data into `Pkg::Config` when jobs depend on `pl:fetch`.

This is probably not the complete refactoring that this code deserves, but it should be a reasonable first step in making our configuration more deterministic and lowering the cognitive load required to understand how Packaging's configuration works.

### CODE REVIEW

**Comment by mckern** [comment]
(Spoiler alert: I would not merge this until we have tried to ship something with it. There's no useful spec tests around the Rake tasks.)

File: README.md [hunk 1]
```diff
@@ -732,7 +732,7 @@ files:
 
 * **pl:ips**
 
-    Invoke package:ips, but do so after pl:fetch and pl:load_extras, which load
+    Invoke package:ips, but do so after pl:fetch, which load
```
Comment (mwaggett) [inline README.md]: nitpick, but `load` => `loads`

**Comment by shrug** [comment]
💯 to the idea, needs some thorough testing to find what blows up

Does it need tests?

**Comment by mckern** [comment]
Yeah, it probably needs tests. Will add a new commit.

**Comment by stahnma** [comment]
@shrug figure out where to schedule this in for shipment/testing etc.

**Comment by underscorgan** [comment]
There are a few tests I want to run before we merge this, looking this afternoon:
- [ ] Vanagon Build
- [ ] EZBake Build
- [ ] uber_ship up to the actual shipping

**Comment by underscorgan** [comment]
This fails with ezbake builds, not sure where it got to where it's requiring PE_VER for foss builds:

```
16:07:26 Copying file to BUILD_PROPERTIES
16:07:26 Copying file to PROJECT_BUNDLE
16:07:30 [pl_deb COW=base-jessie-i386.cow] $ /bin/bash /tmp/hudson6412161733033150877.sh
16:07:31 ~/workspace/puppetserver-packaging-2017-02-07-16-07-17-2.7.3.stable.SNAPSHOT.2017.02.07T1607/command/pl_deb COW=base-jessie-i386.cow/project ~/workspace/puppetserver-packaging-2017-02-07-16-07-17-2.7.3.stable.SNAPSHOT.2017.02.07T1607/command/pl_deb COW=base-jessie-i386.cow
16:07:31 Cloning into 'git_repo'...
16:07:48 Note: checking out '901e0529493fc318e289f0db378787b6ac8c07c2'.
16:07:48 
16:07:48 You are in 'detached HEAD' state. You can look around, make experimental
16:07:48 changes and commit them, and you can discard any commits you make in this
16:07:48 state without impacting any branches by performing another checkout.
16:07:48 
16:07:48 If you want to create a new branch to retain commits you create, you may
16:07:48 do so (now or later) by using -b with the checkout command again. Example:
16:07:48 
16:07:48   git checkout -b new_branch_name
16:07:48 
16:07:49 ~/workspace/puppetserver-packaging-2017-02-07-16-07-17-2.7.3.stable.SNAPSHOT.2017.02.07T1607/command/pl_deb COW=base-jessie-i386.cow/project/git_repo ~/workspace/puppetserver-packaging-2017-02-07-16-07-17-2.7.3.stable.SNAPSHOT.2017.02.07T1607/command/pl_deb COW=base-jessie-i386.cow/project ~/workspace/puppetserver-packaging-2017-02-07-16-07-17-2.7.3.stable.SNAPSHOT.2017.02.07T1607/command/pl_deb COW=base-jessie-i386.cow
16:07:49 EZBAKE PROJECT NAME: puppetserver
16:07:49 cd ext
16:07:53 cd -
16:07:53 EZBAKE PROJECT NAME: puppetserver
16:07:53 ** Invoke pl:deb (first_time)
16:07:53 ** Invoke package:tar (first_time)
16:07:53 ** Invoke clean (first_time)
16:07:53 ** Execute clean
16:07:53 rm -rf ext/files pkg
16:07:53 ** Execute package:tar
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/redhat/puppetserver.spec
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/redhat/puppetserver.service
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/debian/puppetserver.default_file
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/debian/puppetserver.init_script
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/debian/puppetserver.service_file
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/puppetserver.logrotate.conf
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/puppetserver.logrotate-legacy.conf
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/puppetserver.tmpfiles.conf
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/redhat/init
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/debian/changelog
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/debian/control
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/debian/docs
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/debian/postrm
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/debian/preinst
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/debian/prerm
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/debian/postinst
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/debian/rules
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/bin/puppetserver
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/cli/reload
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/cli/start
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/cli/irb
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/cli/foreground
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/cli/stop
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/cli/gem
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/cli/ruby
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/default
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/ext/ezbake-functions.sh
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/Makefile
16:07:54 Generated: /tmp/pkg09X8ti/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607/install.sh
16:07:56 ** Execute pl:deb
16:07:56 rake aborted!
16:07:56 Requires PE_VER be set!
16:07:56 /var/lib/jenkins/workspace/puppetserver-packaging-2017-02-07-16-07-17-2.7.3.stable.SNAPSHOT.2017.02.07T1607/command/pl_deb COW=base-jessie-i386.cow/project/git_repo/ext/packaging/lib/packaging/util.rb:55:in `check_var'
16:07:56 /var/lib/jenkins/workspace/puppetserver-packaging-2017-02-07-16-07-17-2.7.3.stable.SNAPSHOT.2017.02.07T1607/command/pl_deb COW=base-jessie-i386.cow/project/git_repo/ext/packaging/tasks/deb.rake:116:in `block (2 levels) in <top (required)>'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/task.rb:205:in `call'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/task.rb:205:in `block in execute'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/task.rb:200:in `each'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/task.rb:200:in `execute'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/task.rb:158:in `block in invoke_with_call_chain'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/task.rb:151:in `invoke_with_call_chain'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/task.rb:144:in `invoke'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/application.rb:116:in `invoke_task'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/application.rb:94:in `block (2 levels) in top_level'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/application.rb:94:in `each'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/application.rb:94:in `block in top_level'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/application.rb:133:in `standard_exception_handling'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/application.rb:88:in `top_level'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/application.rb:66:in `block in run'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/application.rb:133:in `standard_exception_handling'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/rake/application.rb:63:in `run'
16:07:56 /usr/local/rvm/rubies/ruby-1.9.3-p551/bin/rake:32:in `<main>'
16:07:56 Tasks: TOP => pl:deb
16:07:56 Wrote /var/lib/jenkins/workspace/puppetserver-packaging-2017-02-07-16-07-17-2.7.3.stable.SNAPSHOT.2017.02.07T1607/command/pl_deb COW=base-jessie-i386.cow/project/git_repo/pkg/puppetserver-2.7.3.stable.SNAPSHOT.2017.02.07T1607.tar.gz
16:07:56 Build step 'Execute shell' marked build as failure
16:07:56 Build step 'Groovy Postbuild' marked build as failure
16:07:56 Archiving artifacts
16:07:59 Finished: FAILURE
```

**Comment by mckern** [comment]
That's a start! Will dig in this week.

**Comment by underscorgan** [comment]
@mckern trying to gather build params for you, will let you know when it's up

**Pull request closed without merging**